### PR TITLE
[refactor - 1.21.0] Deduplicate valory package

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR 

- Ports `http`, `ledger_api` and `contract_api` protocols as valory packages 
- Remove the tendermint protocol as a package and utilises sync to download it whenever necessary

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -114,6 +114,18 @@ basepython = python3.10
...
+
+[testenv:sync]
+skipsdist = True
+usedevelop = True
+basepython = python3
+deps =
+    ipfshttpclient==0.8.0a2
+commands =
+    python -m pip install --no-deps file://{toxinidir}/plugins/aea-cli-ipfs
+    python -m aea.cli packages sync
```
Comment (DavidMinarsch) [inline tox.ini:126]: Should not be needed, `open-aea` is the most upstream repo. Please remove.

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -64,6 +64,7 @@ jobs:
           sudo apt-get autoremove
           sudo apt-get autoclean
           pip install tox
+          tox -e sync
```
Comment (DavidMinarsch) [inline .github/workflows/workflow.yml:67]: Please remove the syncs - see other comments.

File: .gitignore [hunk 3]
```diff
@@ -126,6 +126,9 @@ joined-path
 
 packages/default_author/
 
+packages/valory/protocols/tendermint
```
Comment (DavidMinarsch) [inline .gitignore:129]: Why? We don't need this protocol in this repo.

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: packages/valory/protocols/contract_api/__init__.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 # ------------------------------------------------------------------------------
 #
-#   Copyright 2022 fetchai
+#   Copyright 2022 valory
```
Comment (Karrenbelt) [inline packages/valory/protocols/contract_api/__init__.py:4]: inconsistent with `Valory AG` used in other places, but I suppose author name cannot include a white space either

File: packages/valory/connections/test_libp2p/tests/test_dht.py [hunk 5]
```diff
@@ -369,10 +339,3 @@ def test_randomized_default_message_exchange(self, message):
         """Test randomized default message strategy"""
 
         assert self.send_message_via_random_multiplexer_pair(message)
-
-    @settings(deadline=2000)
-    @given(st.builds(lambda kw: TendermintMessage(**kw), tendermint_message_strategy))
-    def test_randomized_tendermint_message_exchange(self, message):
-        """Test randomized tendermint message strategy"""
-
-        assert self.send_message_via_random_multiplexer_pair(message)
```
Comment (Karrenbelt) [inline packages/valory/connections/test_libp2p/tests/test_dht.py:378]: lets move this to open-autonomy

File: packages/valory/protocols/contract_api/__init__.py (hunk 4 above)
Comment (Karrenbelt) [reply packages/valory/protocols/contract_api/__init__.py:4]: could maybe fix this in update copyright headers, that is to say map "valory" to "Valory AG"

File: packages/valory/connections/test_libp2p/tests/test_dht.py (hunk 5 above)
Comment (angrybayblade) [reply packages/valory/connections/test_libp2p/tests/test_dht.py:378]: Please open an issue for this

File: packages/valory/protocols/contract_api/__init__.py (hunk 4 above)
Comment (angrybayblade) [reply packages/valory/protocols/contract_api/__init__.py:4]: I will address this on 343

Comment (angrybayblade) [reply packages/valory/protocols/contract_api/__init__.py:4]: Ohk, so this is generated by the protocol generator so this is the right format.

**Comment by Karrenbelt** [comment]
failure windows py3.10
```
    def test_check_versions_negative_cannot_parse_version() -> None:
        """Test the check_versions - negative case, cannot parse version."""
        with mock.patch("sys.stdout", new_callable=StringIO) as mock_stdout:
            with mock.patch("subprocess.check_output", return_value=b""):
                check_versions()
            stdout = mock_stdout.getvalue()
>       assert (
            "Warning: cannot parse 'go' version from command: ['go', 'version']." in stdout
        )
E       assert "Warning: cannot parse 'go' version from command: ['go', 'version']." in ''
```

I've tried to address in #340

**Review by Karrenbelt — APPROVED** [review]
opened issue to port the test, can happen alongside the tendermint protocol: https://github.com/valory-xyz/open-aea/issues/344

File: .gitignore [hunk 6]
```diff
@@ -139,7 +140,6 @@ packages/fetchai/connections/p2p_libp2p/libp2p_node/libp2p_node
 !plugins/aea-ledger-fetchai/tests/data/dummy_contract/build
 
 deploy-image/.env
-deploy-image/packages/valory
```
Comment (DavidMinarsch) [inline .gitignore:142]: why the change?

**Review by DavidMinarsch — DISMISSED** [review]

File: .gitignore (hunk 6 above)
Comment (angrybayblade) [reply .gitignore:142]: We don't copy packages to the image directory anymore

**Comment by Adamantios** [comment]
This PR also addresses #232, since it includes changes introduced in https://github.com/valory-xyz/open-autonomy/pull/1109:
https://github.com/valory-xyz/open-aea/blob/f63b2534ac661a17f34cebdbb44f66f68cb84248/packages/valory/connections/ledger/ledger_dispatcher.py#L278-L305

**Review by Adamantios — DISMISSED** [review]

**Review by Karrenbelt — DISMISSED** [review]

**Review by DavidMinarsch — APPROVED** [review]
lgtm
